compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -942,43 +982,32 @@ private boolean isObjectMemberQualifier(SyntaxKind tokenKind) {
     private void parseTopLevelQualifiers(List<STNode> qualifiers) {
         while (isTopLevelQualifier(peek().kind)) {
             STToken qualifier = consume();
-            if (isNodeWithSyntaxKindInList(qualifiers, qualifier.kind)) {
-                updateLastNodeInListOrAddInvalidNodeToNextToken(qualifiers, qualifier,
-                        DiagnosticErrorCode.ERROR_DUPLICATE_QUALIFIER, qualifier.text());
-            } else {
-                qualifiers.add(qualifier);
-            }
+            qualifiers.add(qualifier);
         }
     }
 
     private void parseTypeDescQualifiers(List<STNode> qualifiers) {
         while (isTypeDescQualifier(peek().kind)) {
             STToken qualifier = consume();
-            if (isNodeWithSyntaxKindInList(qualifiers, qualifier.kind)) {
```

Comment:
Removed considering the language complications. 
**Example 1:**
At top level something like below is valid.
```bal
isolated isolated object { } a = b;
```
- Here we have duplicate qualifiers and we don't know if it results in an syntax error until equal sign is parsed. 
- This forces us to remove the duplicate restriction from `parseTopLevelQualifiers` and move that validation to the end.
- Since duplicate restriction was removed from `parseTopLevelQualifiers`, for the consistency I removed it `parseTypeDescQualifiers`, `parseObjectMemberQualifiers` so that duplication validation takes place at the parsing end in all places, which I think is cleaner. WDYT?

**Example 2:**
At both top level and type descriptor level something like below is a valid type. (recursively using distinct type desc).
```bal
distinct distinct distinct student ;
```
 For now I have excluded `distinct` as a top level/ type desc level qualifier with following check. But, in future we can consider adding it to the qualifier list and parse it recursively in `parseTypeDescriptorInternal`.

```bal
case DISTINCT_KEYWORD: // class-def, distinct-type-desc
                nextNextToken = getNextNextToken();
                // distinct-type-desc can occur recursively.
                // e.g. `distinct distinct student` is a valid type descriptor
                // Treat distinct as a top level qualifier only with class definition.
                switch (nextNextToken.kind) {
                    case CLIENT_KEYWORD:
                    case READONLY_KEYWORD:
                    case ISOLATED_KEYWORD:
                    case CLASS_KEYWORD:
                        return true;
                    default:
                        return false;
                }
```

Reply:
I see. ok